### DISCUSSION TITLE
Parse error in customConstraint(): Expected a constraint

### DISCUSSION THREAD

**Discussion by matthewkooshad** [simolus3/drift#2476] [Q&A] [answered]
greetings, I just now updated from an old version to the latest, and here's an issue i need help with please.

my stuff table has
IntColumn get otherId => integer().nullable().customConstraint('NULLABLE REFERENCES others(id)')();

1 │ NULLABLE REFERENCES others(id)
  │ ^^^^^^^^

line 1, column 1: Parse error in customConstraint(): Expected a constraint (primary key, nullability, etc.)

how do i resolve this warning? thanks!

**Comment by simolus3** [comment, marked as answer]
`NULLABLE` is not a valid constraint in SQL. I don't fully understand why sqlite3 would accept it, I guess it thinks its part of the type if it's the first constraint.

The new feature was designed to help spot those errors earlier, so I'd say it's working as intended :) You can fix the warning by removing `NULLABLE` from `customConstraint` - columns are nullable by default in SQL, so there's nothing more you need to do.

**Reply by matthewkooshad** [reply]
ok, i changed those from
 IntColumn get otherId => integer().nullable().customConstraint('NULLABLE REFERENCES others(id)')();
to
 IntColumn get otherId => integer().nullable().customConstraint('REFERENCES others(id)')();

i also have some examples like:
IntColumn get otherId => integer().customConstraint('NOT NULL REFERENCES others(id)')();

should i remove the "not null" on such examples?
it complains differently saying: `others` could not be found in any import.

**Reply by matthewkooshad** [reply]
i changed from
IntColumn get otherId => integer().customConstraint('NOT NULL REFERENCES others(id)')();
to
IntColumn get otherId => integer().customConstraint('NOT NULL REFERENCES Others(id)')();

capitalizing that table name in the references constraint, and it doesn't complain on that one now. i think that's correct. but please let me know your thoughts on that. i didn't know sql was case sensitive.

**Reply by simolus3** [reply]
> i didn't know sql was case sensitive.

It's not, so this sounds like a bug in drift. If you can share the definition of the two tables, I'd be happy to take a look at what's going on here. I couldn't reproduce this issue in an existing test.

**Reply by matthewkooshad** [reply]
right now, the problem is not showing. i just went through and consistently adjusted all of the references in constraints to use lower case. and also, i had one table where i should have used _ to separate the pascal case table name when referencing it in a constraint. i'll follow-up later if an error resurfaces on this. thanks!

**Comment by matthewkooshad** [comment]
i've ran the `dart run build_runner build --delete-conflicting-outputs` a few times, and i started to see the complaint again on:
`IntColumn get otherId => integer().customConstraint('NOT NULL REFERENCES Others(id)')();`
